Parser/pegen/pegen.c

```diff
@@ -310,8 +310,12 @@ get_error_line(char *buffer, int is_file)
         newline = strchr(buffer, '\n');
     }
 
+    while (newline && *newline == '\n') {
+        newline -= 1;
+    }
+
     if (newline) {
-        return PyUnicode_DecodeUTF8(buffer, newline - buffer, "replace");
+        return PyUnicode_DecodeUTF8(buffer, newline - buffer + 1, "replace");
```

Comment:
This may work but it feels horrible. If the buffer has just `"\n"` then newline may point *before* the buffer. Honestly the whole function feels weird. How would you describe it in words? The original version seems to do

- If `is_file` is set, return the buffer minus the last newline;
- else, return the buffer up to the first newline, if any.

The new semantics is a bit weirder, it's something like

- If `is_file` is set, return the buffer minus any trailing newlines;
- else, return the buffer up to the first newline, if any (i.e., unchanged).

The only acceptable version of the loop would seem to be
```
if (is_file && newline) {
    while (newline > buffer && newline[-1] == '\n')
        --newline;
}
```
and then the return could revert to the original.

Reply:
It seems that the version of the loop that you suggest solves the problem and plays nicely with the rest of the code.

Reply:
> The new semantics is a bit weirder, it's something like

Yeah, I agree with this but seems that if we want to play the same that the old parser, this logic must exist somewhere. I am not sure if there is a more elegant way to distribute it.